iceoryx_utils/include/iceoryx_utils/posix_wrapper/timer.hpp

```diff
@@ -321,6 +321,36 @@ class Timer
 };
 
 } // namespace posix
+
+// IOX-#337 Alternate Timer to replace the POSIX timer
+namespace clock
+{
+class DeadlineTimer
+{
+  public:
+    /// @brief Constructor
+    /// @param[in] timeToWait duration until the timer expires
+    DeadlineTimer(const iox::units::Duration timeToWait) noexcept;
+
+    /// @brief Checks if the timer has expired compared to its start time
+    /// @return Is the elapsed time larger than duration set
+    bool hasExpired() noexcept;
+
+    /// @brief reinitializes the starting time for the timer to the current time
+    void reset() noexcept;
+
+    /// @brief returns the time duration in ms available before the timer expires
+    iox::units::Duration remainingTime() noexcept;
```

Comment:
I'm missing some test to verify the correctness of the remaining time.

Reply:
Yes I have added the test for the remainingTime